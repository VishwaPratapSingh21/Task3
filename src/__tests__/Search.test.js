import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Search from '../components/SearchQuery';

describe('Search component',()=>{
    const wrapper = shallow(<Search handleSearch={jest.fn()}/>)
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });

    it('renders a form', () => {
        expect(wrapper.find('form')).toHaveLength(1);
        expect(wrapper.find('form').simulate('submit', {preventDefault}))
        expect(preventDefault).toBeCalled();
        expect(wrapper.find('[name="q"]').simulate('change', {target:{name:'q', value:'lunch'}}))
    });
})