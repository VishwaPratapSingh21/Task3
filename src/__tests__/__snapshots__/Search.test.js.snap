// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders properly 1`] = `
<div
  className="col-md-3"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        name="q"
        onChange={[Function]}
        placeholder="Search for categories "
        type="search"
        value=""
      />
      <span
        className="input-group-btn"
      >
        <button
          className="btn btn-secondary pr-4"
          type="submit"
        >
          <i
            className="fa fa-search"
          />
        </button>
      </span>
    </div>
  </form>
</div>
`;
